File: 1138-grumpy-bookstore-owner/grumpy-bookstore-owner.cpp
class Solution {
public:
    int maxSatisfied(vector<int>& c, vector<int>& g, int mi) {
        int n = c.size();
        int bc = 0;
        
        for (int i = 0; i < n; ++i) {
            if (g[i] == 0) {
                bc += c[i];
            }
        }
        
        int es = 0;
        for (int i = 0; i < mi; ++i) {
            if (g[i] == 1) {
                es += c[i];
            }
        }
        
        int mxs = es;
        for (int i = mi; i < n; ++i) {
            if (g[i] == 1) {
                es += c[i];
            }
            if (g[i - mi] == 1) {
                es -= c[i - mi];
            }
            mxs = max(mxs, es);
        }
        
        return bc + mxs;
    }
};

File: 1138-grumpy-bookstore-owner/grumpy-bookstore-owner.go
func maxSatisfied(c []int, g []int, mi int) int {
    n:=len(c)
    bc:=0
    for i:=0 ;i<n ; i++{
        if g[i]==0{
            bc+=c[i]
        }
    }
    es:=0
    for i:=0 ; i<mi ; i++{
        if g[i]==1{
            es+=c[i]
        }
    }
   mxs :=es
   for i:=mi ;i<n; i++{
    if g[i]==1{
        es+=c[i]
    }
    if g[i-mi]==1{
        es-=c[i-mi]
    }
    if es>mxs{
        mxs=es
    }
   }
return bc+mxs

}